pkg/kubernetes/api/core/v1/pod/build.go

```diff
@@ -0,0 +1,251 @@
+/*
+Copyright 2019-2020 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pod
+
+import (
+	"github.com/openebs/dynamic-localpv-provisioner/pkg/kubernetes/api/core/v1/container"
+	"github.com/openebs/dynamic-localpv-provisioner/pkg/kubernetes/api/core/v1/volume"
+	errors "github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	// k8sNodeLabelKeyHostname is the label key used by Kubernetes
+	// to store the hostname on the node resource.
+	k8sNodeLabelKeyHostname = "kubernetes.io/hostname"
+)
+
+// Builder is the builder object for Pod
+type Builder struct {
+	pod  *Pod
+	errs []error
+}
+
+// NewBuilder returns new instance of Builder
+func NewBuilder() *Builder {
+	return &Builder{pod: &Pod{object: &corev1.Pod{}}}
+}
+
+// WithTolerationsForTaints sets the Spec.Tolerations with provided taints.
+func (b *Builder) WithTolerationsForTaints(taints ...corev1.Taint) *Builder {
+
+	tolerations := []corev1.Toleration{}
+	for i := range taints {
+		var toleration corev1.Toleration
```

Comment:
Can this be declared outside the for loop? Values of struct seem to be over-ridden each the loop runs

Reply:
@RealHarshThakur - couldn't catch it. Which struct is being over-ridden?

the outside variable is for `tolerations`, but inside (loop var) is converting a taint to a `toleration` and appending to the list of tolerations.

Reply:
I meant the inner variable `toleration` can be declared outside the for loop. It can only be declared once, every time the loop runs, fields of `toleration` struct would be over-ridden, so it should work.

Reply:
ah. ok. it is an optimization in case there are no taints, then there is no need to declare that variable.